src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionsHandler.cs

```diff
@@ -39,62 +37,30 @@ public CodeActionsHandler(ICodeFixService codeFixService, ICodeRefactoringServic
             // Filter out code actions with options since they'll show dialogs and we can't remote the UI and the options.
             codeActions = codeActions.Where(c => !(c is CodeActionWithOptions));
 
-            var clientSupportsWorkspaceEdits = true;
-            if (clientCapabilities?.Experimental is JObject clientCapabilitiesExtensions)
-            {
-                clientSupportsWorkspaceEdits = clientCapabilitiesExtensions.SelectToken("supportsWorkspaceEdits")?.Value<bool>() ?? clientSupportsWorkspaceEdits;
-            }
-
             var result = new ArrayBuilder<object>();
             foreach (var codeAction in codeActions)
             {
-                // If we have a codeaction with a single applychangesoperation, we want to send the codeaction with the edits.
-                var operations = await codeAction.GetOperationsAsync(cancellationToken).ConfigureAwait(keepThreadContext);
-                if (clientSupportsWorkspaceEdits && operations.Length == 1 && operations.First() is ApplyChangesOperation applyChangesOperation)
-                {
-                    var workspaceEdit = new LSP.WorkspaceEdit { Changes = new Dictionary<string, LSP.TextEdit[]>() };
-                    var changes = applyChangesOperation.ChangedSolution.GetChanges(solution);
-                    var changedDocuments = changes.GetProjectChanges().SelectMany(pc => pc.GetChangedDocuments());
+                // Note that we can pass through the params for this
+                // request (like range, filename) because between getcodeaction and runcodeaction there can be no
+                // changes on the IDE side (it will requery for codeactions if there are changes).
```

Comment:
What is the "it" here? I'm not sure I follow this comment.

Reply:
I agree the comment is confusing, it wasn't added in this PR but might be worth fixing up.  I think 'it' could either be the client or the server.  Before running code actions on the server, we re-query for the up to date code action.  And if there are changes on the client side, the client will also re-query for the list of code actions.